src/orange/orangeinp/detail/PolygonUtils.hh

```diff
@@ -119,6 +83,84 @@ is_same_orientation(Orientation a, Orientation b, bool degen_ok = false)
     return (a == b);
 }
 
+//---------------------------------------------------------------------------//
+/*!
+ * Functor for calculating orientation with a tolerance for collinearity.
+ *
+ * Collinearity is based on a supplied absolute tolerance. For three ordered
+ * points a, b, c, point b is collinear if the displacement, d, is less than
+ * the absolute tolerance.
+ *
+ *              b
+ *             . .
+ *           .  .  .
+ *         .    .    .
+ *       .      . d    .
+ *     .  t     .        .
+ *   a . . . . . . . . . . c
+ *
+ * The displacement is calculated as follows.
+ *
+ * Let:
+ * u = b - a
+ * v = c - a
+ *
+ * In 2D, the cross product is can be written as,
+ *
+ * u x v = |u| |v| sin(t),
```

Comment:
LaTeX-like formatting should be used for math equations in comments:
```suggestion
 * \f[
 *  u \times v = |u| |v| \sin(t),
 * \f]
```
Or maybe even "\vec u" and "\vec v".  
See src/celeritas/field/MagFieldEquation.hh for examples.